inference-engine/src/mkldnn_plugin/nodes/common/emitter.cpp

```diff
@@ -182,11 +198,25 @@ void jit_emitter::prepare_table() {
     }
 }
 
-void jit_emitter::emit(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs,
+void jit_emitter::emit(const std::vector<size_t> &in_idxs, const std::vector<size_t> &out_idxs,
                        const std::vector<size_t> &pool_vec_idxs, const std::vector<size_t> &pool_gpr_idxs) {
-    emitter_preamble(in_vec_idxs, pool_vec_idxs, pool_gpr_idxs);
+    emitter_preamble(in_idxs, out_idxs, pool_vec_idxs, pool_gpr_idxs);
+
+    emit_impl(in_idxs, out_idxs, pool_vec_idxs, pool_gpr_idxs);
 
-    emit_impl(in_vec_idxs, out_vec_idxs, pool_vec_idxs, pool_gpr_idxs);
+    emitter_postamble();
+}
+
+void jit_emitter::emit(const std::vector<size_t> &in_idxs, const std::vector<size_t> &out_idxs,
+                       const std::shared_ptr<const emitter_context> &emit_context,
+                       const std::vector<size_t> &pool_vec_idxs, const std::vector<size_t> &pool_gpr_idxs) {
+    emitter_preamble(in_idxs, out_idxs, pool_vec_idxs, pool_gpr_idxs);
+
+    if (emit_context) {
+        emit_impl(in_idxs, out_idxs, pool_vec_idxs, pool_gpr_idxs, emit_context.get());
+    } else {
+        emit_impl(in_idxs, out_idxs, pool_vec_idxs, pool_gpr_idxs);
```

Comment:
If emit_impl is not overridden in derivative class when emit_impl from base class will be called (which in fact do nothing: https://github.com/openvinotoolkit/openvino/blob/e158be0c4311cc95a0246c879334c7a44a4836f4/inference-engine/src/mkldnn_plugin/nodes/common/emitter.h#L90 
So seems like we need to do the following:
1) emit(... , ctx) should call only emit_impl(... , ctx) and have default value for ctx
2) emit(...) should call emit(... , nullptr)
3) remove emit_impl(...) at all
What do you think?

Reply:
Agree with below understanding:
> 1. emit(... , ctx) should call only emit_impl(... , ctx) and have default value for ctx
        I think you mean emit_impl(... , ctx = nullptr), not emit(... , ctx = nullptr). Otherwise if call emit(...), we don't know 
        execute  emit(...) or emit(..., ctx=nullptr), it's ambiguous.
> 2. emit(...) should call emit(... , nullptr)
        should call emit_impl(... , nullptr)?
> 3. remove emit_impl(...) at all and keep only  emit_impl(... , ctx)
       Yes, I checked global emitter. There are no emit_impl() base function. So we can remove or change flexibly in cpu 
       emitter.

Reply:
Yeah, you are right. In case we want to keep version with default ctx value we need to implement your approach. 
Seems it also makes sense to make emit_impl pure.

Reply:
done by [6a7a844](https://github.com/openvinotoolkit/openvino/pull/3567/commits/6a7a8445a45481ed9513b5a3bc7632230e6835bd)